ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java

```diff
@@ -83,6 +84,15 @@ public void shouldCreateConnectionOnScanStart() throws Exception {
     }
 
     @Test
+    public void shouldCreateConnectionWithCredentialsOnScanStart() throws Exception {
+        when(xaConnectionFactoryMock.createXAConnection(anyString(), anyString())).thenReturn(xaConnectionMock);
```

Comment:
Why is that horrible? I don't know any easier way to mimic such scenario. Just telling the connection factory to return a specific connection instance, if that method is called with any string arguments.